test/test-project/gulpfile.js

```diff
@@ -48,5 +68,21 @@ gulp.task('test1', () => {
     // TODO(justinfagnani): allow filtering of files before writing
     .pipe(gulp.dest('build/unbundled'));
 
-  return mergeStream(bundled, unbundled);
+  return Promise.all([waitFor(bundled), waitFor(unbundled)]).then(() => {
+    return Promise.all([
+      addServiceWorker({
```

Comment:
we'll just need to find a way to include the buildRoot in each config, and do it in a way that the user can still customize the call on top of this. Agreed that this should be treated as out of scope for now.

Lets keep that waiting logic outside of the function for now. I'm pushing a new change that waits for each individual buildPhase to create the service worker, instead of coupling them both together. Let me know what you think.